slam/alarm_tree_view.py

```diff
@@ -246,7 +252,15 @@ def send_action(self, acknowledged: Optional[bool] = None, enabled: Optional[boo
                     values_to_send = self.create_config_values_for_action(alarm, enabled, acknowledged)
                     if enabled is not None and enabled != alarm.is_enabled():
                         # Changes to enabled status go to the regular topic
-                        self.kafka_producer.send(self.topic, key=f"config:{alarm_path}", value=values_to_send)
+
+                        # empty topic string means this is 'All' topic tree-vew and doesn't have a valid kafka topic,
+                        # so grab the destination topic from the alarm's path.
+                        if self.topic == "":
+                            self.kafka_producer.send(
+                                alarm_path.split("/")[1], key=f"config:{alarm_path}", value=values_to_send
+                            )
```

Comment:
I think setting the topic like this needs to be done for acknowledgments too in the lines right below this, as well as in the  `alarm_table_view`.